PEP 728: Fix formatting and code highlighting issues.

PIG208 opened the pull request:
This update addresses @hugovk [comments](https://github.com/python/peps/pull/3441#pullrequestreview-1871689915).

<!-- readthedocs-preview pep-previews start -->
----
📚 Documentation preview 📚: https://pep-previews--3659.org.readthedocs.build/

<!-- readthedocs-preview pep-previews end -->

JelleZijlstra reviewed (approved):
Thanks for the fix, and thanks @hugovk for the review!

hugovk commented:
Thanks! Looks good!

Whilst I'm looking at it now (and sorry for not reviewing on the first pass, but these are minor things :), there's a few places with a bit too much indentation, which shows up as quoted text:

<img width="519" alt="image" src="https://github.com/python/peps/assets/1324225/3686532d-1568-483f-849f-d1a67baa2f5c">

Plus one code block (line 417) that's shown as regular text.

Shall we fix these too?

By the way, good work with the movie examples! Much easier to follow than foo and bar.

PIG208 commented:
Thanks for the reviews! I fixed the code blocks and some other issues. Also updated the type consistency part to make the edits stand out. I plan on doing another PR focusing on content changes after getting more feedback from the discussion post.

hugovk commented:
Thanks! These too:

<img width="809" alt="image" src="https://github.com/python/peps/assets/1324225/f94b9864-8b20-405e-b9a8-51fd4945067e">

<img width="799" alt="image" src="https://github.com/python/peps/assets/1324225/2d5b7c48-6236-4143-8757-1352cf1603c9">

<img width="819" alt="image" src="https://github.com/python/peps/assets/1324225/c5c5c807-381f-41e5-b638-beec65655e0b">

JelleZijlstra commented:
@hugovk I think those are essential, they are (modified) quotes from the existing spec.
